src/openklant2/types/resources/partij.py

```diff
@@ -0,0 +1,103 @@
+from typing import Literal, NotRequired, Required
+
+from pydantic import TypeAdapter
+from typing_extensions import TypedDict
+
+from openklant2.types.common import Adres, ForeignKeyRef
+from openklant2.types.iso_639_2 import LanguageCode
+
+
+class Contactnaam(TypedDict):
+    voorletters: str
+    voornaam: str
+    voorvoegselAchternaam: str
+    achternaam: str
+
+
+# Note this is polymorphic, concrete types below
+class CreatePartijDataBase(TypedDict):
+    nummer: NotRequired[str]
+    interneNotitie: NotRequired[str]
+    digitaleAdressen: Required[list[ForeignKeyRef] | None]
+    voorkeursDigitaalAdres: Required[ForeignKeyRef | None]
+    rekeningnummers: Required[list[ForeignKeyRef] | None]
+    voorkeursRekeningnummer: Required[ForeignKeyRef | None]
+    voorkeurstaal: LanguageCode
+    indicatieActief: bool
+    indicatieGeheimhouding: bool
+    correspondentieadres: NotRequired[Adres]
+    bezoekadres: NotRequired[Adres | None]
+
+
+class PartijIdentificatiePersoon(TypedDict):
+    contactnaam: Required[Contactnaam | None]
+
+
+class PartijIdentificatieContactpersoon(TypedDict):
+    contactnaam: Required[Contactnaam | None]
+    werkteVoorPartij: ForeignKeyRef
+
+
+class PartijIdentificatieOrganisatie(TypedDict):
+    naam: Required[str | None]
+
+
+class CreatePartijPersoonData(CreatePartijDataBase):
+    soortPartij: Literal["persoon"]
+    partijIdentificatie: PartijIdentificatiePersoon
+
+
+class CreatePartijContactpersoonData(CreatePartijDataBase):
+    soortPartij: Literal["contactpersoon"]
+    partijIdentificatie: PartijIdentificatieContactpersoon
+
+
+class CreatePartijOrganisatieData(CreatePartijDataBase):
+    soortPartij: Literal["organisatie"]
+    partijIdentificatie: PartijIdentificatieOrganisatie
+
+
+class Partij(TypedDict):
+    uuid: Required[str]
+    nummer: NotRequired[str]
+    interneNotitie: NotRequired[str]
+    digitaleAdressen: Required[list[ForeignKeyRef] | None]
+    voorkeursDigitaalAdres: Required[ForeignKeyRef | None]
+    rekeningnummers: Required[list[ForeignKeyRef] | None]
+    voorkeursRekeningnummer: Required[ForeignKeyRef | None]
+
+    voorkeurstaal: LanguageCode
```

Comment:
I thought the distinction between `Required` and `NotRequired` is exhaustive, but this is neither (same for the attributes below)?